tests/garage/torch/algos/test_sac.py

```diff
@@ -0,0 +1,146 @@
+import gym
+import numpy as np
+import torch
+from torch.nn import functional as F  # NOQA
+from torch import nn as nn
+
+from garage.envs import normalize
+from garage.envs.base import GarageEnv
+from garage.experiment import LocalRunner, run_experiment
+from garage.replay_buffer import SimpleReplayBuffer
+from garage.torch.algos import SAC
+from garage.torch.policies import TanhGaussianMLPPolicy
+from garage.torch.q_functions import ContinuousMLPQFunction
+
+
+from unittest import mock
+from unittest.mock import MagicMock
+
+
+class _MockDistribution(object):
+
+    def __init__(self, action):
+        self._action = action
+
+    def rsample(self):
+        return self._action
+
+    def log_prob(self, unused_sample):
+        return torch.Tensor([10.])
+
+
+class DummyActorPolicy(object):
+
+    def __init__(self,
+                action=1.):
+        self._action = action
+
+    def __call__(self, observation):
+        action = torch.Tensor([self._action])
+        return _MockDistribution(action)
+
+    def action(self, unused_observation):
+        action = torch.Tensor([self._action], dtype=torch.float32)
+        return action
+
+    def parameters(self):
+        return torch.zeros(5)
+
+
+class DummyCriticNet():
+
+    def __init__(self):
+        pass
+
+    def parameters(self):
+        return torch.zeros(5)
+
+    def __call__(self, observation, actions):
+        # Biggest state is best state.
+        value = torch.max(observation, dim=-1).values
+        # Biggest action is best action.
+        q_value = torch.max(actions, axis=-1).values
+        ret = value + q_value
+        return ret
+
+
+class TestSacLosses():
```

Comment:
seems like there's a lot of repetitive tests here -- maybe you can to define a `setup_method` instead?

note that pytest is perfectly happy to test ordinary functions rather than unittest-style classes. see: https://docs.pytest.org/en/latest/getting-started.html and https://docs.pytest.org/en/latest/fixture.html#fixtures

also: there's no need for a trailing ()